### DISCUSSION TITLE
Is it really async ? ... becoz "requests" is not async ... it should use httpx (which is really async!)

### DISCUSSION THREAD

**Discussion by manatlan** [xtekky/gpt4free#2358] [Q&A] [answered]
Perhaps I'm wrong ... perhaps async is done via greenlet or something else ...

But "requests" is really not async compliant (it lock the thread during http exchange) ... while httpx, aiohttp are a better option.
Shouldn't be the time to migrate from "requests" to "httpx" ?

**Comment by hlohaus** [comment, marked as answer]
Our utilization of requests is sporadic. The majority of providers employ either aiohttp or curl_cffi. Asynchronous functionality is only available when utilizing the AsyncClient.
